replica/server.go

```diff
@@ -188,14 +189,32 @@ func (s *Server) PrevStatus() (State, Info) {
 
 func (s *Server) Stats() *types.Stats {
 	r := s.r
-	return &types.Stats{
-		RevisionCounter: r.revisionCache,
-		ReplicaCounter:  int64(r.peerCache.ReplicaCount),
+	var revisionCache int64
+	var replicaCount int64
+
+	revisionCache = 0
+	replicaCount = 0
+	if r != nil {
+		revisionCache = r.revisionCache
+		replicaCount = int64(r.peerCache.ReplicaCount)
+	}
+
+	stats1 := &types.Stats{
+		RevisionCounter: revisionCache,
+		ReplicaCounter:  replicaCount,
 	}
+	return stats1
 }
 
 func (s *Server) GetUsage() (*types.VolUsage, error) {
-	return s.r.GetUsage()
+	if s.r != nil {
+		return s.r.GetUsage()
+	}
+	return &types.VolUsage{
```

Comment:
why this change? shouldn't we return error in this case?

Reply:
error seems to be ignored in the code, which is why just returning error may not be sufficient. It needs nil handling in the callers.
Added a TODO for nil handling